lib/ProductOpener/API.pm

```diff
@@ -833,4 +837,103 @@ sub customize_response_for_product ($request_ref, $product_ref, $fields_comma_se
 	return $customized_product_ref;
 }
 
+=head2 process_auth_header ( $request_ref, $r )
+
+Using the Authorization HTTP header, check if we have a valid user.
+
+=head3 Parameters
+
+=head4 $request_ref (input)
+
+Reference to the request object.
+
+=head4 $r (input)
+
+Reference to the Apache2 request object
+
+=head3 Return value
+
+None
+
+=cut
+
+sub process_auth_header ($request_ref, $r) {
+	my $token = _read_auth_header($request_ref, $r);
+	unless ($token) {
+		return;
+	}
+
+	my $access_token;
+	eval {$access_token = verify_access_token($token);};
+	my $error = $@;
+	if ($error) {
+		$log->info('Access token invalid', {token => $token}) if $log->is_info();
+	}
+
+	unless ($access_token) {
+		add_error(
```

Comment:
I only added `process_auth_header` to v3 for now, anyways. Do you think we should look into adding it to older API versions, too? I was thinking that it's the API version most likely to be used by applications that will easily integrate OAuth/OIDC, anyways.

Reply:
I think we should add it to v2 because we want the situation to improve for many and v3 is considered a not stable version of the API (we may change things without warning).
Some already developed app using v2 API might want to change the authentication mode. The open food facts mobile app for example use v2 and v3 at the same time.

Reply:
Should now work for API v2, including image upload.